pkg/cri/config/config_unix.go

```diff
@@ -96,7 +96,7 @@ func DefaultConfig() PluginConfig {
 		ImageDecryption: ImageDecryption{
 			KeyModel: KeyModelNode,
 		},
-		EnableCDI:                false,
+		EnableCDI:                true,
```

Comment:
As this branch is targeting "v2" (which has some breaking changes in config), I wonder if we can change the config to `disable_cdi` (`enable_cdi: false` is awkward; even more so in some cases with Golang using `false` as default value for booleans, which can be problematic with (un)marshaling).

Reply:
Technically there is no problem to rename it, but it can confuse users as they already used to `EnableCDI` option.

Looking at the existing config options, I can see that there are other Enable* options:
```
pkg/cri/config/config_unix.go:		EnableSelinux:        false,
pkg/cri/config/config_unix.go:		EnableTLSStreaming:   false,
pkg/cri/config/config_unix.go:		EnableCDI:                true,
pkg/cri/config/config_unix.go:		EnableUnprivilegedPorts:  true,
pkg/cri/config/config_unix.go:		EnableUnprivilegedICMP:   true,
pkg/cri/config/config_windows.go:		EnableTLSStreaming:  false,
```

Are we going to rename all of them or `EnableCDI` is in some way unique in this sense?

Reply:
V2 may provide use the opportunity to rename all of those that are enabled by default. I think it makes for a better user-experience, and I know we implemented "deprecation warnings" for config options, which we included in 1.6 and 1.7 releases to give users a heads-up.

If it helps transitioning, we could keep a struct with the old names that, when used, print a deprecation warning (something like _"enable_cdi is deprecated: CDI is now enabled by default. Use "disable_cdi" instead to disable CDI"_)


All of this is my $0.02c; I haven't worked on / been involved in discussions around deprecation of other options, so perhaps others have their opinions, but thought I'd leave it as a suggestion to consider 🤗 ❤️

Reply:
Thank you for the detailed explanations. In my opionion we should either rename all of the Enable* options and produce a deprecation warning or keep them as they are to avoid user confusion.

BTW, `DisableCDI: false` doesn't look much better than `EnableCDI: false` to me. `EnableCDI: true` is more clear and that's what this PR is proposing to have as a default.

Reply:
> BTW, DisableCDI: false doesn't look much better than EnableCDI: false to me. EnableCDI: true is more clear and that's what this PR is proposing to have as a default.

Yes, I guess it depends on how these are used; my assumption here was that the default wouldn't have to be shown anywhere (if nothing is set, then no need to include the option in your config, so it can be omitted).

Reply:
Can we merge this PR as is and then rename all Enable* config options in a different PR? This would be fair from my point of view.

@mikebrow @AkihiroSuda @fuweid WDYT?

Reply:
I don't think enable/disable should have a hard and fast rule..

This one is a bit special.  Memory serves the enable false was a short term way to disable.  We'll want to deprecate the flag as we head into scenarios where CDI is required for k8s/CRI can add a todo here if like to cover those KEPs.  Admins will be able to manage which CDI plugins are installed. Switching from default false to default true makes sense before deprecation.

Reply:
then as DRA and CDI device plugins in k8s become ubiquitous we can delete the flag.

Reply:
Ah, yes, if it's required then removing the option entirely makes sense. (Don't know details about the other options); thanks!

Reply:
@mikebrow thank you for the detailed explanations and LGTM! I've added TODO as suggested.

Reply:
I'm not sure if the deprecation criteria(GA graduation for any of the k8s features depending on CDI) is good enough. Feel free to suggest a better criteria.

Reply:
looks good.. links show which keps to track..  There are others when looking outside k8s use of CRI.. but this is fine I think since atm k8s manages the CRI..